src/cross_toml.rs

```diff
@@ -0,0 +1,194 @@
+#![doc = include_str!("../docs/cross_toml.md")]
+
+use crate::errors::*;
+use crate::Target;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+/// Build environment configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossBuildEnvConfig {
+    volumes: Option<Vec<String>>,
+    passthrough: Option<Vec<String>>,
+}
+
+/// Build configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossBuildConfig {
+    env: Option<CrossBuildEnvConfig>,
+    xargo: Option<bool>,
+    target: Option<String>,
+}
+
+/// Target configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossTargetConfig {
+    passthrough: Option<Vec<String>>,
+    volumes: Option<Vec<String>>,
+    xargo: Option<bool>,
+    image: Option<String>,
+    runner: Option<String>,
+}
+
+/// Wrapper struct for mappings from [`Target`] to [`CrossTargetConfig`]
+///
+/// This is used to circumvent that serde's flatten and field aliases
```

Comment:
this can be a `// TODO:` instead of docstring

Reply:
also, is this really applicable here?

```rust
pub struct CrossToml {
    #[serde(rename = "target")]
    targets: HashMap<Target, CrossTargetConfig>
    build: Option<CrossBuildConfig>,
}
```

would work no?

Reply:
Oh, true, I don't know why I thought that the wrapper struct would be necessary.